apps/api/src/app/topics/use-cases/add-subscribers/add-subscribers.use-case.ts

```diff
@@ -16,19 +18,28 @@ interface ISubscriberGroups {
 @Injectable()
 export class AddSubscribersUseCase {
   constructor(
+    private createTopic: CreateTopicUseCase,
     private searchByExternalSubscriberIds: SearchByExternalSubscriberIds,
     private topicSubscribersRepository: TopicSubscribersRepository,
     private topicRepository: TopicRepository
   ) {}
 
   async execute(command: AddSubscribersCommand): Promise<Omit<ISubscriberGroups, 'subscribersAvailableToAdd'>> {
-    const topic = await this.topicRepository.findTopicByKey(
+    let topic = await this.topicRepository.findTopicByKey(
       command.topicKey,
       command.organizationId,
       command.environmentId
     );
+
     if (!topic) {
-      throw new NotFoundException(`Topic with key ${command.topicKey} not found in current environment`);
+      const createTopicCommand = CreateTopicCommand.create({
```

Comment:
Should we check that the command.topicKey is valid? Like not null or empty string or a certain min length?

Reply:
Good points!
If `topicKey` is not sent (empty string) when doing the call the URL would be the kind of `POST /api/topics//subscribers` and the App would return a 404 Not Found as it doesn't exist. If they send a null that will be converted to a string by their client so we will be ok. Also we have the validation in the command that would throw a 400 error: https://github.com/novuhq/novu/blob/b5cf0dcb26f2e938b25a3c4799a794d7f0d5e4ea/apps/api/src/app/topics/use-cases/add-subscribers/add-subscribers.command.ts#L9

I am not too much worried about the length string and so as they have the option to rename the topic with the API.